core/shared/src/main/scala/zio/internal/FiberContext.scala

```diff
@@ -812,7 +812,8 @@ private[zio] final class FiberContext[E, A](
           while (iterator.hasNext()) {
             val child = iterator.next()
 
-            child.parentFiber = self.parentFiber
+            if (self.parentFiber ne null) child.parentFiber = self.parentFiber
+            else Fiber.track(child)
```

Comment:
I think we need an `onDone` here that clears the child from the tracked set.

Reply:
Yes It is part of `track`

Reply:
Ah right